docs/layering.md

```diff
@@ -0,0 +1,369 @@
+
+This would be helpful when IT needs to provide the images or there is a need to add the same system image to multiple hosts. 
+Instead of making the changes on each host which will have different versions of software or configs you can create an image which then you can rebase your systems on. We are using the [fcos-derivation-example](https://github.com/coreos/fcos-derivation-example) to start this process.
+
+Let's clone our example code and enter the directory.
+
+```bash
+[core@tutorial ~]$ git clone git@github.com:coreos/fcos-derivation-example.git && cd fcos-derivation-example
+```
+
+Examine the Dockerfile and notice the custom binary and where we are installing strace.
+
+```bash
+$ vi Dockerfile
+```
+
+Let's add `cachefilesd` to the install command.
+
+That would make the last line look like:
+```dockerfile
+RUN rpm-ostree install strace cachefilesd && rpm-ostree cleanup -m
+```
+
+Time to build the image. In this example we are using podman.
+```
+$ podman build -t localhost/my-custom-fcos .
+```
+
+### Inspect the content of the derived image
+At this point we have an OCI image. You can test to see if it actually has the layered packages. 
+
+Let's get into the container.
+```
+$ podman run -it localhost/my-custom-fcos /bin/bash
+```
+
+Run stace inside the container. You should see the following:
+```bash
+$ strace -- version 5.15
+
+Copyright (c) 1991-2021 The strace developers <https://strace.io>.
+This is free software; see the source for copying conditions.  There is NO
+warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+Optional features enabled: stack-trace=libdw stack-demangle m32-mpers mx32-mpers secontext
+```
+
+Which tells us we have successfully installed `strace` on Fedora CoreOS.
+
+Next let's see if cachefilesd is there.
+```bash
+$ cachefilesd --version     
+```
+
+The output should look like the following:
+```
+cachefilesd version 0.10.10
+```
+
+Now that we confirmed our packages are there, we can get out of the container:
+```bash
+$ exit
+```
+
+### Push the OCI image to a container registry
+
+In order to use our image, we need to push it to an image repository that is accessible to our hosts.
+
+In this example we will use `quay.io`. But this can be any compatible container image repository.
+
+You will need to create an account on [quay.io](quay.io).
+
+Login to quay.io from your terminal:
+
+```bash
+$ podman login quay.io
+```
+
+Push the previously built image:
+
+```bash
+$ podman push localhost/my-custom-fcos quay.io/<YOURUSER>/my-custom-fcos
+```
+
+### Rebase the OS to the derived image
+
+For this step we go back to our Fedora CoreOS VM console.
+
+We need to stop the Zincati service so that auto-updates won't interfere with our manual operations.
+
+```
+[core@tutorial ~]$ sudo systemctl stop zincati.service
+```
+
+Next, we call `rpm-ostree` to rebase our system using the image we just pushed to quay.
+
+```bash
+[core@tutorial ~]$ sudo rpm-ostree rebase --experimental \
```

Comment:
This is fine as is, I don't want you to rewrite this now.  Let's get this in and iterate.

But I think the story is more compelling if we run this rebase as part of the butane - and even add higher level sugar for it - https://github.com/coreos/enhancements/blob/main/os/coreos-layering.md#use-of-coreos-diskboot-images